### PR TITLE
added support for mounted drives via unc paths.

### PR DESCRIPTION
**Pull request by erinxocon**
Path('R:\repo\').resolve() seems to resolve to `\\computername\repos\repo' when using a mounted drive.  The path in a form of R:\repo\ is technically an absolute path already so it doesn't need converting.

### CODE REVIEW

**Comment by ahendry2688** [comment]
@erinxocon Thank you very much for your help Erin! I'm trying to figure out how to pull your changes and rerun my tests to verify that it resolves my original issue.

**Comment by erinxocon** [comment]
@ahendry2688 Just pull from master and then checkout the `unc-paths` branch.  I do see an error in CI, but think it might just be transient as it didn't fail the other windows test.

**Comment by ahendry2688** [comment]
@erinxocon Do you mean the following?:

`git pull origin master`, then
`git checkout -b unc-paths`.

**Comment by erinxocon** [comment]
@ahendry2688 indeed.  Then ensure pipenv is uninstalled and set it up as a development dependency by doing `pip install -e .`

**Comment by ahendry2688** [comment]
@erinxocon Thank you Erin. Just checking. I'm trying to run the tests right now. Is there a way I can run these in parallel? @uranusjr said I could use `-n` flag, but I don't see that flag option in `pytest --help` or online docs. Can I run

`pipenv run pytest tests -n 24`

if, e.g., I have 24 (virtual) cores?

**Comment by ahendry2688** [comment]
@erinxocon Thanks for your help. I'm still getting errors. Here are the commands I issued:

`pip uninstall pipenv
pip install pytest-xdist
git pull origin master
git checkout -b unc-paths
pip install -e .
pipenv run pytest tests -n 24`

And here is the output (see attached .txt file):

[pytest_output_PR2331.txt](https://github.com/pypa/pipenv/files/2087235/pytest_output_PR2331.txt)

**Comment by ahendry2688** [comment]
@erinxocon Corrected using `-n auto`. Errors still remain.

[pytest_output_PR2331_using_n_auto.txt](https://github.com/pypa/pipenv/files/2087246/pytest_output_PR2331_using_n_auto.txt)

**Comment by ahendry2688** [comment]
@erinxocon If it helps, here is my output from running `pytest --fixtures`.

[pytest_fixtures_output.txt](https://github.com/pypa/pipenv/files/2087254/pytest_fixtures_output.txt)

**Comment by ahendry2688** [comment]
@erinxocon Installed `pytest-fixtures` and `pytest-pypi`. Rerunning tests, but now getting many more errors.
[pytest_output_PR2331_run002.txt](https://github.com/pypa/pipenv/files/2087351/pytest_output_PR2331_run002.txt)

**Comment by techalchemy** [comment]
All you should need to run is `set PYPI_VENDOR_DIR=".\tests\pypi\" && set PYTHONIOENCODING="utf-8" && cd pipenv && pip install -e . && pipenv install --dev && pipenv run pytest -v -n 4 --ignore=".\\pipenv\\patched" --ignore=".\\pipenv\\vendor" tests`

**Comment by ahendry2688** [comment]
@erinxocon Even with techalchemy's suggestions, my tests still report errors. I will continue to look into this. Please let me know if you do not want my text file outputs anymore and I will stop sending them.

[pytest_output_PR2331_run003.txt](https://github.com/pypa/pipenv/files/2087427/pytest_output_PR2331_run003.txt)

**Comment by techalchemy** [comment]
@erinxocon there was a concern around it accurately writing to a Pipfile if the pipfile itself is stored on a share, I don't currently have a way to test this  scenario but I suspect if we aren't calling `normalize_path` _everywhere_ it might be a source of problems.  We might need to brainstorm a unit test for an edge case or something

**Comment by erinxocon** [comment]
@techalchemy so far this is the _only_ way I can get it to write to the pipfile, right now with my mounted drives.  Do you know how to mount shares on app veyor?  I'll have to do some research to see about setting up a test like that

**Comment by ahendry2688** [comment]
Thank you both so much for all your hard work! If it's any consolation, my company uses all shared drives to collaborate with colleagues as we have yet to purchase a private GitHub repo. This PR would be a huge help to us, and I'm hoping to other users as well. Thank you!

**Comment by techalchemy** [comment]
@erinxocon when you get a chance can you uncomment the appveyor matrix?

**Comment by erinxocon** [comment]
@techalchemy done, sorry bout that, was messing around with app veyor.  Squashed the commits out of existence.

File: pipenv/project.py [hunk 1]
```diff
@@ -51,7 +51,11 @@
 def _normalized(p):
     if p is None:
         return None
-    return normalize_drive(str(Path(p).resolve()))
+    loc = Path(p)
+    if loc.is_absolute():
+        return normalize_drive(str(loc))
+    else:
+        return normalize_drive(str(loc.resolve()))
```
Comment (techalchemy) [inline pipenv/project.py]: based on our findings in the other PR we've been hammering away at, can you make this second part into:
```python
    else:
        try:
            loc = loc.resolve()
        except OSError:
            loc = loc.absolute()
        return normalize_drive(str(loc))
```

In order to handle Ramdisks...

Comment (erinxocon) [reply pipenv/project.py]: Done

**Comment by techalchemy** [comment]
:cake:
